### PR TITLE
 [Role/Permission] User accounts changes (3 of 4)

### PR DESCRIPTION
**Pull request by ridz1208**
This adds the Role infrastructure to LORIS. This is one of 4 PRs which will ultimately result in assigning roles to users. each role has affiliated permissions and automatically checks the proper permission when selected. This change will be mainly visible in the user_accounts module.

PART 3: add all the front-end code to use the new infrastructure

in order to test, the following SQL inserts can be used:
```
INSERT INTO role (Name, Label) VALUES ('coordinator', 'Coordinator');
INSERT INTO role (Name, Label) VALUES ('data', 'Data-Entry');
INSERT INTO role (Name, Label) VALUES ('upload', 'Data-upload');


INSERT INTO role_permission_rel VALUES ((SELECT RoleID FROM role WHERE Name='coordinator'), (SELECT permID FROM permissions WHERE code='send_to_dcc'));
INSERT INTO role_permission_rel SELECT RoleID,permID FROM role JOIN permissions WHERE Name='data' AND code IN ('data_dict_view','data_entry','data_dict_edit','candidate_parameter_view');
INSERT INTO role_permission_rel SELECT RoleID,permID FROM role JOIN permissions WHERE Name='upload' AND code IN ('video_upload','media_write','media_read');
```

Original PR by @taracampbell : https://github.com/aces/Loris/pull/2642
Updated/Standardized/DAO full version : https://github.com/ridz1208/Loris/pull/9

### CODE REVIEW

**Comment by ridz1208** [comment]
Pending Merger of:
- https://github.com/aces/Loris/pull/3511 
- https://github.com/aces/Loris/pull/3537
- https://github.com/aces/Loris/pull/3538

File: modules/user_accounts/ajax/getRolePermissionData.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    return $roles;
+}
+
+/**
+ * Fetches the permissions from the database
+ * Each permission is set to checked if the user being edited has that permission,
+ * and set to enabled if the editing user has the permission
+ *
```
Comment (PapillonMcGill) [inline modules/user_accounts/ajax/getRolePermissionData.php]: missing ending dot

File: modules/user_accounts/ajax/getRolePermissionData.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+function editingSelf()
+{
+    $userEditingID = \User::singleton()->getUsername();
+    $userToEditID  = $_REQUEST['identifier'];
+
+    if ($userEditingID == $userToEditID) {
+        return true;
+    } else {
+        return false;
+    }
```
Comment (PapillonMcGill) [inline modules/user_accounts/ajax/getRolePermissionData.php]: multiple return

**Comment by driusan** [comment]
Closing for the same reason as #3538

**Pull request closed without merging**